### DISCUSSION TITLE
reducing confusion between dvc remote and git remote

### DISCUSSION THREAD

**Discussion by skshetry** [treeverse/dvc#5939] [General]
I have seen too many people getting confused over `dvc` vs `git` remote.

Looks like this should be handled by `dvc` after all. 

We could either:
1. Change terminology from `remote` to `storage` or `cache`, similarly change commands:
```
dvc storage add s3 s3://bucket/
``` 

We can keep using `remote` as an alias.

2. Try to see if the remote url is a git remote based on heuristics (eg: if domain name is github for example) and thrown a error and a hint for what to do.

... More ideas?

cc @jorgeorpinel

**Comment by shcheklein** [comment]
It's a good one. The only thing to keep in mind that we use our "remotes" not only as storages - e.g. `remote://` notation for external deps/outs.

**Comment by skshetry** [comment]
`storage://` make more sense to me as well. Again, we can keep `remote://` as an alias if we don't want to break compatibility.

```console
$ dvc add storage://dvc-test/123
```

**Comment by efiop** [comment]
How would the other commands look like? E.g. push/pull?
```
dvc push --storage mys3
```
?

Should we change the terminology for the .dvc/cache too?

**Comment by jorgeorpinel** [comment]
> Change terminology from remote to storage... keep using remote as an alias

➕1️⃣ !!! I would maybe just use the term "cache storage" sometimes in key places of the docs.

>  dvc push --storage mys3

@efiop -s/--storage should work, yes.

> change the terminology for the .dvc/cache too?

What do you mean? As in .dvc/storage? I think that's a separate question, could be considered but term "cache" isn't a source of confusion AFAI. Again, I think we can associate the concepts effectively in key docs.

**Comment by efiop** [comment]
> $ dvc add storage://dvc-test/123

@skshetry Btw, this doesn't look very good. See https://github.com/iterative/dvc/issues/3920 , it feels like this use case is a "workspace" case. And this might be where `storage` might be confusing, as it is too ambiguous, whereas `remote` is a storage in dvc-specific format (our cache format), which is similar to `git remote`s.

**Comment by efiop** [comment]
> What do you mean? As in .dvc/storage? I think that's a separate question, could be considered but term "cache" isn't a source of confusion AFAI. Again, I think we can associate the concepts effectively in key docs.

@jorgeorpinel We have actually heard questions about it too :slightly_smiling_face: It is really closer to `.git/objects`.
